src/cpu/aarch64/brgemm/brgemm.hpp

```diff
@@ -0,0 +1,240 @@
+/// @param dt_b Data type of B vector can be: f32
+/// @note
+///     Data type of matrix C f32 data type
+/// @param transA Specifies the form of A used in the matrix multiplication
+///        'false' - A is not transposed, 'true' - A is transposed
+/// @param layout Specifies whether two-dimensional array storage is row-major
+///        (brgemm_row_major) or column-major (brgemm_col_major).
+/// @param alpha Specifies the scalar alpha
+/// @param beta Specifies the scalar beta
+/// @param LDA Specifies the leading dimension of matrix A.
+///        LDA must be at least max(1, N)
+/// @param LDC Specifies the leading dimension of matrix C.
+///       LDC must be at least max(1, N)
+/// @param M Specifies the number of rows of the matrix A and C.
+/// @param N Specifies the number of columns of the matrix A and C.
+///
+status_t DNNL_API brdgmm_desc_init(brgemm_t *brg, cpu_isa_t isa,
+        brgemm_batch_kind_t type, impl::data_type_t dt_a,
+        impl::data_type_t dt_b, bool transA, brgemm_layout_t layout,
+        float alpha, float beta, dim_t LDA, dim_t LDC, dim_t M, dim_t N,
+        const brgemm_strides_t *strides = nullptr);
+
+/// Adds post-operations to BRGEMM descriptor
+///
+/// @param brg Output BRGEMM descriptor
+/// @param attr Primitive attributes (can be nullptr). Specifies post-ops
+///     operations
+/// @param dst_md Specifies the memory descriptor of the destination tensor,
+///     needed for binary postops to determine broadcast type, as well as to
+///     determine dst data type.
+/// @param LDD Specifies the leading dimension of matrix D
+///        LDD must be at least max(1, N)
+/// @param dt_bias Specifies the data type Bias
+///     Can be u8, s8, s32, bf16 or fp32
+///
+status_t DNNL_API brgemm_desc_set_postops(brgemm_t *brg,
+        const primitive_attr_t *attr, const memory_desc_t *dst_md, int LDD,
+        impl::data_type_t dt_bias = impl::data_type::undef);
+
+/// Adds BRGEMM attributes to BRGEMM descriptor
+///
+/// @param brg Output BRGEMM descriptor
+/// @param brgattr Specifies kernel attributes and hints: virtual padding,
+///     maximum batch size, kernel loop order etc.
+///
+status_t DNNL_API brgemm_desc_set_attr(
+        brgemm_t *brg, const brgemm_attr_t &brgattr);
+
+/// Generates a BRGEMM kernel based on descriptor
+///
+/// @param brg_kernel Output BRGEMM kernel
+/// @param brg BRGEMM descriptor
+///
+status_t DNNL_API brgemm_kernel_create(
+        brgemm_kernel_t **brg_kernel, const brgemm_t &brg);
+
+/// Destroys a BRGEMM kernel
+///
+/// @param brg_kernel BRGEMM kernel
+///
+status_t DNNL_API brgemm_kernel_destroy(brgemm_kernel_t *brg_kernel);
+
+/// Execute BRGEMM kernel (brgemm_addr version)
+///
+/// @note
+///     Only BRGEMM kernel will be executed even if post-ops are added to BRGEMM
+///     descriptor
+///
+/// @note
+///     In row major mode matrix B (matrix A for column major) is expected to be
+///     in a VNNI-friendly format, which requires 4 consecutive elements of K
```

Comment:
Still a few more intel specific comments.

Reply:
Hello @cfRod, thank you for sharing this review comment.

We are using the same granularity as VNNI granularity on our SVE for the general data types. Hence, we have maintained the same name throughout. Changing that would require many changes to functions and variables as well. Since the functionality and meaning is same (for data type granularity) we have maintained the same.